### PR TITLE
PE-4033: add country and to s checkbox to top up modal

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/7tjnvefofsdj8
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/6s9o573fqnq28

### CODE REVIEW

File: lib/services/turbo/payment_service.dart [hunk 1]
```diff
@@ -112,6 +112,16 @@ class PaymentService {
 
     return PaymentModel.fromJson(jsonDecode(result.data));
   }
+
+  Future<List<String>> getSupportedCountries() async {
+    final result = await httpClient.get(
+      url: '$turboPaymentUri/v1/countries',
+    );
+
+    final countries = jsonDecode(result.data) as List<dynamic>;
```
Comment (karlprieb) [inline lib/services/turbo/payment_service.dart:121]: why `dynamic`? don't we know that the returned data will be a list of strings?

File: lib/services/turbo/payment_service.dart [hunk 2]
```diff
@@ -147,6 +157,12 @@ class DontUsePaymentService implements PaymentService {
     required String currency,
   }) =>
       throw UnimplementedError();
+
+  @override
+  Future<List<String>> getSupportedCountries() {
+    // TODO: implement getSupportedCountries
```
Comment (karlprieb) [inline lib/services/turbo/payment_service.dart:163]: why this comment?

Comment (thiagocarvalhodev) [reply lib/services/turbo/payment_service.dart:163]: That's a code snipped from flutter to add missing methods. I'll remove

File: lib/services/turbo/payment_service.dart (hunk 1 above)
Comment (karlprieb) [reply lib/services/turbo/payment_service.dart:121]: But if you're saying it's string on the line below, why just not set as `List<String>`?

**Review by karlprieb — APPROVED** [review]

File: lib/services/turbo/payment_service.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/services/turbo/payment_service.dart:121]: At runtime, it returns a List<dynamic> which we can't infer the type `String` :(.

**Review by karlprieb — DISMISSED** [review]

**Review by karlprieb — DISMISSED** [review]

File: lib/l10n/app_en.arb
Comment (matibat) [inline lib/l10n/app_en.arb:1]: We'll need a Task to put these keys in Localizely if we haven't

File: lib/services/turbo/payment_service.dart [hunk 3]
```diff
@@ -112,6 +112,14 @@ class PaymentService {
 
     return PaymentModel.fromJson(jsonDecode(result.data));
   }
+
+  Future<List<String>> getSupportedCountries() async {
+    final result = await httpClient.get(
+      url: '$turboPaymentUri/v1/countries',
+    );
```
Comment (matibat) [inline lib/services/turbo/payment_service.dart:119]: Doesn't it need to run retries?

File: lib/turbo/topup/blocs/payment_form/payment_form_bloc.dart [hunk 4]
```diff
@@ -15,18 +15,53 @@ class PaymentFormBloc extends Bloc<PaymentFormEvent, PaymentFormState> {
...
+              state.priceEstimate,
+              _expirationTimeInSeconds(turbo.maxQuoteExpirationDate),
+            ));
+
+            final supportedCountries = await turbo.getSupportedCountries();
+
+            emit(
+              PaymentFormLoaded(
+                state.priceEstimate,
+                _expirationTimeInSeconds(turbo.maxQuoteExpirationDate),
```
Comment (matibat) [inline lib/turbo/topup/blocs/payment_form/payment_form_bloc.dart:30]: This line may throw after successfully getting the supported countries. Would two calls to the same method return different results (i.e. is it pure)?

File: lib/turbo/topup/views/topup_payment_form.dart [hunk 5]
```diff
@@ -322,6 +385,98 @@ class TurboPaymentFormViewState extends State<TurboPaymentFormView> {
...
+        label: appLocalizationsOf(context).nameOnCard,
+        isFieldRequired: true,
+        useErrorMessageOffset: true,
+        validator: (s) {
+          String valid = s?.replaceAll(RegExp(r'[^a-zA-Z\s]'), '') ?? '';
+          _nameController.text = valid;
+          _nameController.selection =
+              TextSelection.collapsed(offset: valid.length);
+
+          setState(() {});
```
Comment (matibat) [inline lib/turbo/topup/views/topup_payment_form.dart:403]: Should be wrapping the variables in the state that we're setting

File: lib/turbo/topup/views/topup_payment_form.dart [hunk 6]
```diff
@@ -585,9 +756,8 @@ class QuoteRefreshWidgetState extends State<QuoteRefreshWidget> {
               return Row(
                 mainAxisAlignment: MainAxisAlignment.center,
                 children: [
-                  // TODO: localize
                   Text(
-                    'Quote updates in ',
+                    appLocalizationsOf(context).quoteUpdatesIn,
```
Comment (matibat) [inline lib/turbo/topup/views/topup_payment_form.dart:760]: May benefit from ARB plurals (maybe tech debt)

File: pubspec.lock [hunk 7]
```diff
@@ -87,8 +87,8 @@ packages:
     dependency: "direct main"
     description:
       path: "."
-      ref: "v1.3.1"
-      resolved-ref: f45178502c62f97a9d300495e0c5a14668aed896
+      ref: PE-4033-add-country-and-to-s-checkbox-to-top-up-modal
```
Comment (matibat) [inline pubspec.lock:90]: Reminder to bump this later on

**Review by matibat — APPROVED** [review]
Approved with comments - didn't review the tests

**Pull request merged**